src/components/views/dialogs/ChatCreateOrReuseDialog.js

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2017 Vector Creations Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import sdk from '../../../index';
+import dis from '../../../dispatcher';
+import MatrixClientPeg from '../../../MatrixClientPeg';
+import DMRoomMap from '../../../utils/DMRoomMap';
+import AccessibleButton from '../elements/AccessibleButton';
+import Unread from '../../../Unread';
+import classNames from 'classnames';
+import createRoom from '../../../createRoom';
+
+export default class CreateOrReuseChatDialog extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this._onNewDMClick = this._onNewDMClick.bind(this);
+        this.dispatcherRef = dis.register(this._onAction.bind(this));
+    }
+
+    componentWillUnmount() {
+        dis.unregister(this.dispatcherRef);
+    }
+
+    _onAction(payload) {
+        switch(payload.action) {
+            case 'view_room':
+                this.props.onFinished(true);
```

Comment:
Similar to the discussion in #riot-dev, I don't particularly like using view-room dispatches for this: I know there's a modal dialog being displayed at the time, but seems entirely possible that something else will decide to view a different room while the modal is being displayed, at which point the modal goes away. In essence, the thing that causes the modal to go away should be the user interaction with the dialog finishing, not the currently viewed room changing.